### PR TITLE
initial work towards applying limits to block helpers

### PR DESCRIPTION
**Pull request by tteltrab**
**Related PRs**  
Next step from #79 . 

**What does this PR do?**  
- enable applying functions to nested objects recursively
- improve typing everywhere
- rename truncates to limits
- bump major version because breaking
- add top-level blocks helpers and limits

**Description of Changes**  
Pretty big rewrite of truncate functions (now limits) which allows recursive application, and provides limits for the rest of the blocks.

Seems pretty big from numbers, but ~200 is just renaming, ~200 is from bad diffs, and a lot of it is either tests or repeatedly applying functions.

**What gif most accurately describes how I feel towards this PR?**

### CODE REVIEW

File: packages/blocks/src/lengthHelpers.ts [hunk 1]
```diff
@@ -1,142 +0,0 @@
-import R, {
```
Comment (tteltrab) [inline packages/blocks/src/lengthHelpers.ts:1]: bad diff, this is `limitHelpers` now (not -142/+233, should just be ~+91

File: packages/blocks/src/lengthHelpers.spec.ts [hunk 2]
```diff
@@ -1,88 +0,0 @@
-import {
```
Comment (tteltrab) [inline packages/blocks/src/lengthHelpers.spec.ts:1]: bad diff, this is `limitHelpers.spec.ts` - not -88/+142, should just be ~+54

File: packages/blocks/src/limitHelpers.spec.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      expect.assertions(2);
+      const applied = applyLimitersWithOverrides(
+        fieldsObj,
+        fieldsDefaultOptions,
+        {
+          fields: [identity, { text: truncate }],
+        },
+      );
+
+      // console.log(fieldsDefaultTruncators, fieldsDefaultLimits, applied);
```
Comment (barlock) [inline packages/blocks/src/limitHelpers.spec.ts:136]: Missed this.

File: packages/blocks/src/limitHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+  over,
+  pick,
+  pipe,
+  prop,
+  take,
+  values,
+  view,
+} from 'ramda';
+
+const dot3 = '...';
```
Comment (barlock) [inline packages/blocks/src/limitHelpers.ts:23]:
```suggestion
const dot3 = '…';
```

Didn't want to use the ellipse character? 😛

Comment (barlock) [reply packages/blocks/src/limitHelpers.ts:23]: 〝Smart ellipsis〞?

File: packages/blocks/src/limitHelpers.spec.ts (hunk 3 above)
Comment (tteltrab) [reply packages/blocks/src/limitHelpers.spec.ts]: lol @ it not getting caught because I commented it out. good.

**Pull request merged**